drush/civicrm.drush.inc

```diff
@@ -1577,9 +1578,31 @@ function drush_civicrm_api() {
     civicrm_get_ctype('Individual')
   );
 
-  $result = civicrm_api($entity, $action, $params);
-  if (!empty($result['is_error'])) {
-    drush_set_error('CIVICRM api error', $result['error_message']);
+  switch ($params['version']) {
+    case 3:
+      try {
+        unset($params['version']);
+        $result = civicrm_api3($entity, $action, $params);
+      }
+      catch (CiviCRM_API3_Exception $e) {
+        drush_set_error('CIVICRM api error', $e->getMessage());
+      }
+      break;
+
+    case 4:
+      try {
+        unset($params['version']);
+        $result = civicrm_api4($entity, $action, $params);
+      }
```

Comment:
The PR is all good just noting that depending on the api call the return value from v4 doesn't print anything useful, so e.g. can't really be used for get operations. Long-term is it worth trying to support such a thing from drush?